src/components/Counter.js

```diff
@@ -0,0 +1,29 @@
+import React, {createElement} from 'react';
+
+export class Counter extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {initialValue: this.props.initialValue};
```

Comment:
You don't need to name the actual state field `initialValue` - this is not an initial value, this is a value that is initialized with initial value. So it should be something like: 
`this.state = {value: this.props.initialValue};`